apps/src/templates/sectionProgressV2/InviteToV2ProgressModal.jsx

```diff
@@ -0,0 +1,164 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import {connect} from 'react-redux';
+
+import Button from '@cdo/apps/componentLibrary/button/Button';
+import {Heading2, BodyTwoText} from '@cdo/apps/componentLibrary/typography';
+import {EVENTS} from '@cdo/apps/lib/util/AnalyticsConstants';
+import analyticsReporter from '@cdo/apps/lib/util/AnalyticsReporter';
+import AccessibleDialog from '@cdo/apps/templates/AccessibleDialog';
+import {default as LinkedButton} from '@cdo/apps/templates/Button';
+import i18n from '@cdo/locale';
+
+import styles from './progress-v2-invitation.module.scss';
+
+const newProgressViewGraphic = require('@cdo/static/teacherDashboard/progressOpenBetaAnnouncementGraphic.png');
+
+function InviteToV2ProgressModal({
+  setShowProgressTableV2,
+  sectionId,
+
+  // from redux
+  dateProgressTableInvtationDelayed,
+  hasSeenProgressTableInvite,
+}) {
+  const [invitationOpen, setInvitationOpen] = React.useState(false);
+
+  React.useEffect(() => {
+    // This effect runs only once on mount
+
+    const timeSinceInvitationLastDelayed = () => {
+      const startingDate = new Date(dateProgressTableInvtationDelayed);
+      const today = new Date();
+      const differenceInMilliseconds = today.getTime() - startingDate.getTime();
+      const differenceInDays = differenceInMilliseconds / (1000 * 3600 * 24);
+      return Math.floor(differenceInDays);
+    };
+
+    const showInvitation = () => {
+      const alreadyViewedInvitation = !!hasSeenProgressTableInvite;
+      if (alreadyViewedInvitation) {
+        return false;
+      } else {
+        if (!!dateProgressTableInvtationDelayed) {
+          return timeSinceInvitationLastDelayed() > 3;
+        } else {
+          return true;
+        }
+      }
+    };
+
+    setInvitationOpen(showInvitation());
+  }, [dateProgressTableInvtationDelayed, hasSeenProgressTableInvite]);
+
+  const handleDismiss = () => {
```

Comment:
@lfryemason it is these event handlers that contain the events getting duplicated.